### PR TITLE
Provide feedback based on the buffer.

### PR DESCRIPTION
**Pull request by norisalsaadie**
Closes https://github.com/libero/reviewer/issues/652

### CODE REVIEW

File: src/domain/file/services/file-service.ts [hunk 1]
```diff
@@ -40,6 +43,73 @@ export class FileService {
         }
     }
 
+    async handleMultipartChunk(
+        userId: string,
+        file: File,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        chunk: any,
+        partNumber: number,
+        s3Stuff: PromiseResult<S3.CreateMultipartUploadOutput, AWSError>,
```
Comment (Cooryd) [inline src/domain/file/services/file-service.ts]: Could this param name be a bit more descriptive?

File: src/domain/file/services/file-service.ts [hunk 2]
```diff
@@ -120,62 +190,72 @@ export class FileService {
...
+            const { ETag } = await this.handleMultipartChunk(
+                userId,
+                file,
+                chunk,
+                partNumber,
+                fileUploadManager,
+                pubsub,
+                bytesRead,
+            );
+            parts.push({ ETag, PartNumber: partNumber });
```
Comment (diversemix) [inline src/domain/file/services/file-service.ts]: is this missing `chucks.push(chunk)`?

File: tests/test.utils.ts [hunk 3]
```diff
@@ -64,13 +64,17 @@ export const uploadManuscript = async (submissionId: string): Promise<AxiosRespo
...
+            fileSize: 115,
         },
     };
 
     body.append('operations', JSON.stringify(operations));
     body.append('map', '{ "1": ["variables.file"] }');
-    body.append('1', 'a', { filename: 'a.txt' });
+    body.append(
+        '1',
+        'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
```
Comment (Cooryd) [inline tests/test.utils.ts]: No problem with the functionality here, could probably use some lorem ipsum though so it doesn't look like an accidental key down issue 🤣

File: tests/test.utils.ts [hunk 4]
```diff
@@ -64,13 +64,17 @@ export const uploadManuscript = async (submissionId: string): Promise<AxiosRespo
...
+        { filename: 'a.txt' },
+    );
```
Comment (will-byrne) [inline tests/test.utils.ts]: Is there an integration test that tests the progress upload sends the expected results?

File: src/domain/file/services/file-service.ts (hunk 1 above)
Comment (norisalsaadie) [reply src/domain/file/services/file-service.ts]: Solved

File: tests/test.utils.ts (hunk 4 above)
Comment (norisalsaadie) [reply tests/test.utils.ts]: yep. In wizard.files.test.ts

File: tests/test.utils.ts (hunk 3 above)
Comment (norisalsaadie) [reply tests/test.utils.ts]: haha sure

File: tests/wizard.files.test.ts [hunk 5]
```diff
@@ -74,7 +74,7 @@ describe('Wizard->Files Integration Tests', () => {
     });
 
     // see https://github.com/libero/reviewer-submission/issues/109
-    it('uploads a manuscript file', async () => {
+    it.only('uploads a manuscript file', async () => {
```
Comment (will-byrne) [inline tests/wizard.files.test.ts]: this .only should be removed :D

Comment (diversemix) [reply tests/wizard.files.test.ts]: Good spot!

Comment (Cooryd) [reply tests/wizard.files.test.ts]: ⭐️ Will approve when this is sorted 👍

Comment (diversemix) [reply tests/wizard.files.test.ts]: :drum:

**Pull request merged**